l3kernel/l3str.dtx

```diff
@@ -979,33 +979,45 @@
 %     \str_gput_right:Nn, \str_gput_right:NV, \str_gput_right:Ne, \str_gput_right:Nx,
 %     \str_gput_right:cn, \str_gput_right:cV, \str_gput_right:ce, \str_gput_right:cx
 %   }
-%   Simply convert the token list inputs to \meta{strings}.
+%   Similar to corresponding \pkg{l3tl} base functions, except that
+%   \cs{__kernel_exp_not:w} is replaced with \cs{__kernel_tl_to_str:w}.
 %    \begin{macrocode}
-\group_begin:
-  \cs_set_protected:Npn \@@_tmp:n #1
-    {
-      \tl_if_blank:nF {#1}
-        {
-          \cs_new_protected:cpe { str_ #1 :Nn } ##1##2
-            {
-              \exp_not:c { tl_ #1 :Ne } ##1
-                { \exp_not:N \tl_to_str:n {##2} }
-            }
-          \cs_generate_variant:cn { str_ #1 :Nn }
-            { NV , Ne , Nx , cn , cV , ce , cx }
-          \@@_tmp:n
-        }
-    }
-  \@@_tmp:n
-    { set }
-    { gset }
-    { const }
-    { put_left }
-    { gput_left }
-    { put_right }
-    { gput_right }
-    { }
-\group_end:
+\cs_new_protected:Npn \str_set:Nn #1#2
+  { \__kernel_tl_set:Ne #1 { \__kernel_tl_to_str:w {#2} } }
+\cs_gset_protected:Npn \str_gset:Nn #1#2
+  { \__kernel_tl_gset:Ne #1 { \__kernel_tl_to_str:w {#2} } }
+\cs_new_protected:Npn \str_const:Nn #1#2
+  {
+    \__kernel_chk_if_free_cs:N #1
+    \cs_gset_nopar:Npe #1 { \__kernel_tl_to_str:w {#2} }
```

Comment:
`\__kernel_tl_gset:Ne` (just to be uniform with the others)

Reply:
Like `\tl_const:Nn`, we need `\cs_gset_nopar:Npe` here for scope checking `c` (accomplished by specially patching `\<module>_const:Nn` in `l3debug.def`) rather than `g`. I added some explanation in https://github.com/latex3/latex3/pull/1349/commits/b03e1c02e66b1c5ea2c86f8dda94dd367c55affc.